OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1618,6 +1618,166 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
+                default: throw new Exception("Invalid argument type for arctan2");
+            }
+
+            return ProcStatus.Continue;
+        }
+
+        public static ProcStatus Sqrt(DMProcState state) {
+            DreamValue value = state.Pop();
+
+            switch (value.Type) {
+                case DreamValue.DreamValueType.Float: state.Push(new DreamValue(MathF.Sqrt(value.MustGetValueAsFloat()))); break;
+                case DreamValue.DreamValueType.DreamObject when value == DreamValue.Null: state.Push(new DreamValue(MathF.Sqrt(0.0f))); break;
+                default: throw new Exception("Invalid argument type for sqrt");
+            }
+
+            return ProcStatus.Continue;
+        }
+
+        public static ProcStatus Log(DMProcState state) {
+            DreamValue baseValue = state.Pop();
+            DreamValue value = state.Pop();
+
+            float baseValueFloat = 0.0f;
+
+            switch (baseValue.Type) {
+                case DreamValue.DreamValueType.Float: baseValueFloat = baseValue.MustGetValueAsFloat(); break;
+                case DreamValue.DreamValueType.DreamObject when baseValue == DreamValue.Null: baseValueFloat = 0.0f; break;
+                default: throw new Exception("Invalid argument type for log");
+            }
+
+            switch (value.Type) {
+                case DreamValue.DreamValueType.Float: state.Push(new DreamValue(MathF.Log(value.MustGetValueAsFloat(), baseValueFloat))); break;
+                case DreamValue.DreamValueType.DreamObject when value == DreamValue.Null: state.Push(new DreamValue(MathF.Log(0.0f, baseValueFloat))); break;
+                default: throw new Exception("Invalid argument type for log");
+            }
+
+            return ProcStatus.Continue;
+        }
+
+        public static ProcStatus LogE(DMProcState state) {
+            DreamValue value = state.Pop();
+
+            switch (value.Type) {
+                case DreamValue.DreamValueType.Float: state.Push(new DreamValue(MathF.Log(value.MustGetValueAsFloat()))); break;
+                case DreamValue.DreamValueType.DreamObject when value == DreamValue.Null: state.Push(new DreamValue(MathF.Log(0.0f))); break;
+                default: throw new Exception("Invalid argument type for loge");
+            }
+
+            return ProcStatus.Continue;
+        }
+
+        public static ProcStatus Abs(DMProcState state) {
+            DreamValue value = state.Pop();
+
+            switch (value.Type) {
+                case DreamValue.DreamValueType.Float: state.Push(new DreamValue(MathF.Abs(value.MustGetValueAsFloat()))); break;
+                case DreamValue.DreamValueType.DreamObject when value == DreamValue.Null: state.Push(new DreamValue(0.0f)); break;
+                default: throw new Exception("Invalid argument type for abs");
+            }
+
```

Comment:
`UnsafeGetValueAsFloat()`
```suggestion
            state.Push(new DreamValue(MathF.Abs(value.UnsafeGetValueAsFloat())));
```